src/Loop.php

```diff
@@ -59,155 +59,162 @@ public static function stop()
     /**
      * Defer the execution of a callback.
      *
-     * @param callable $callback The callback to defer.
+     * @param callable(mixed $data, string $watcherIdentifier) $callback The callback to defer.
+     * @param mixed $data Arbitrary data given to the callback function as the $data parameter.
      *
-     * @return string An identifier that can be used to cancel, enable or disable the event.
+     * @return string An identifier that can be used to cancel, enable or disable the watcher.
      */
-    public static function defer(callable $callback)
+    public static function defer(callable $callback, $data = null)
     {
-        return self::get()->defer($callback);
+        return self::get()->defer($callback, $data);
     }
 
     /**
      * Delay the execution of a callback. The time delay is approximate and accuracy is not guaranteed.
      *
-     * @param callable $callback The callback to delay.
+     * @param callable(mixed $data, string $watcherIdentifier) $callback The callback to delay.
```

Comment:
> Hmm, why did you assume this side-effect?

I think I've spent too much time with libevent and libev where watchers are disabled after invocation unless a *persistent* option is given when setting up the watcher. Keeping `on*` watchers active until disabled is definitely the better option.

> FWIW I always liked to have the $watcherId be the first argument

I actually had the watcher ID as the first argument initially, then swapped it to the last argument before committing. I don't really care, so long as it's consistent. If more people feel `$watcherIdentifier` should be first, I have no problem moving it. Many end users of libs won't use watchers directly anyway.